k2/python/k2/fsa_algo.py

```diff
@@ -145,3 +158,137 @@ def arc_sort(fsa: Fsa) -> Fsa:
+            value = a_value.index_select(0, a_arc_map) \
+                    + b_value.index_select(0, b_arc_map)
+            setattr(out_fsa, name, value)
+
+    if hasattr(a_fsa, 'aux_labels'):
+        a_aux_labels = a_fsa.aux_labels
+    else:
+        # this is an acceptor
+        a_aux_labels = a_fsa.labels
+    padding = a_aux_labels.new_zeros(1)
+
+    a_aux_labels = torch.cat((padding, a_aux_labels), dim=0)
+    out_fsa.aux_labels = a_aux_labels.index_select(0, a_arc_map)
+
+    if hasattr(b_fsa, 'aux_labels'):
+        b_aux_labels = b_fsa.aux_labels
+    else:
+        b_aux_labels = b_fsa.labels
+    b_aux_labels = torch.cat((padding, b_aux_labels), dim=0)
+    out_fsa.labels = b_aux_labels.index_select(0, b_arc_map)
+
+    for name, a_value in a_fsa.named_non_tensor_attr():
+        if name not in ('symbols', 'aux_symbols'):
+            setattr(out_fsa, name, a_value)
+
+    for name, b_value in b_fsa.named_non_tensor_attr():
+        if not hasattr(out_fsa, name) and name not in ('symbols',
+                                                       'aux_symbols'):
+            setattr(out_fsa, name, b_value)
+
+    if hasattr(a_fsa, 'aux_symbols'):
+        out_fsa.aux_symbols = a_fsa.aux_symbols
+    elif hasattr(a_fsa, 'symbols'):
+        out_fsa.aux_symbols = a_fsa.symbols
+
+    if hasattr(b_fsa, 'aux_symbols'):
+        out_fsa.symbols = b_fsa.aux_symbols
+    elif hasattr(b_fsa, 'symbols'):
+        out_fsa.symbols = b_fsa.symbols
+
+    return out_fsa
+
+
+def shortest_distance(fsa: Fsa) -> torch.Tensor:
+    '''Return the shortest distance from the start state to the final state
+    in the tropical semiring.
+
+    Note:
+      It uses the opposite sign. That is, It uses `max` instead of `min`.
+
+    Args:
+      fsa:
+        The intpu FSA.
+    Returns:
+      The sum of scores along the best path.
+    '''
+    need_arc_map = True
+    _, arc_map = _k2.shortest_distance(fsa.arcs, need_arc_map)
+    arc_map = arc_map.to(torch.int64)
+
+    return fsa.score[arc_map].sum()
+
+
+def shortest_path(fsa: Fsa) -> Fsa:
+    need_arc_map = True
+    ragged_arc, arc_map = _k2.shortest_path(fsa.arcs, need_arc_map)
+    arc_map = arc_map.to(torch.int64)  # required by index_select
+    out_fsa = Fsa.from_ragged_arc(ragged_arc)
+    for name, value in fsa.named_tensor_attr():
+        setattr(out_fsa, name, value.index_select(0, arc_map))
+    for name, value in fsa.named_non_tensor_attr():
+        setattr(out_fsa, name, deepcopy(value))
+    return out_fsa
+
+
+def project_output(fsa: Fsa) -> Fsa:
+    out_fsa = Fsa.from_ragged_arc(fsa.arcs)
+    out_fsa.labels = fsa.aux_labels
+    for name, value in fsa.named_tensor_attr():
+        setattr(out_fsa, name, value)
+    for name, value in fsa.named_non_tensor_attr():
+        if name != 'symbols':
+            setattr(out_fsa, name, deepcopy(value))
+
+    if hasattr(out_fsa, 'aux_labels'):
+        out_fsa.symbols = out_fsa.aux_labels
```

Comment:
I think this may be a typo: surely should be out_fsa.aux_symbols?  
But since now the labels and aux_labels are the same there is no sense in keeping the aux_labels, so it would make sense to delete out_fsa.aux_symbols and out_fsa.aux_labels.  Anyway maybe the simplest fix right now it just to delete this.

Reply:
Yes, you are right. Will remove this function.

Reply:
Removed.